### DISCUSSION TITLE
Error messaging in admiral: review & rework

### DISCUSSION THREAD

**Discussion by manciniedoardo** [pharmaverse/admiral#2318] [Ideas]
### Background Information

From our 2024 roadmap call, @bundfussr made great points about some of the limitations of our error messaging in admiral - e.g. errors thrown when required variables are missing from `dataset`/`dataset_add` don't indicate in which dataset they are missing. We should carry out a review across our admiral functions, exploring where error messaging can be improved in terms of: verboseness and detail.

Improvement of our error messaging would be a great enhancement to the admiral experience with minimal (no?) breaking changes, perfectly embodying our admiral post-1.0 ethos.

### Definition of Done

- Review of error messages across admiral functions (focus on key ones first) to identify any gaps/confusion. Perhaps with a document outlining all the potential improvements? This document could then be reviewed by the team in order to decide on an action plan.
- Issues created to improve error messaging.

**Comment by manciniedoardo** [comment]
@bundfussr I have assigned you to this issue as this was originally your suggestion - feel free to proceed with this if you like, or unassign yourself if you would not. I also think more than one person can be involved.

**Comment by manciniedoardo** [comment]
Potential idea: can we get a console message when users are using superseded functions?

**Comment by bms63** [comment]
> Potential idea: can we get a console message when users are using superseded functions?

I feel like this should be a separate issue from the error/warning to users.

> @bundfussr I have assigned you to this issue as this was originally your suggestion - feel free to proceed with this if you like, or unassign yourself if you would not. I also think more than one person can be involved.

I think this should be done in a breakout group with some more in-depth discussion.  As this will need some investigation as well as thoughtful discussion.

**Comment by manciniedoardo** [comment]
> I think this should be done in a breakout group with some more in-depth discussion. As this will need some investigation as well as thoughtful discussion.

Agree - honestly by initially assigning @bundfussr I was hoping that you could produce a review document, like the one for the function reword effort, that could form the basis of the work of such a group! It was so helpful that maybe we could replicate this modus operandi.

**Comment by bms63** [comment]
https://cli.r-lib.org/index.html

**Comment by bms63** [comment]
I'm going to move this to a Discussion as this is a large issue that needs to be broken into smaller chunks to tackle.  Daniel has a presentation for us on Wednesday to help orient us from another project?? @ddsjoberg  

https://www.danieldsjoberg.com/ARD-onboarding/05-cli-conditions.html

**Comment by bms63** [comment]
Closing in favor of this current discussion - https://github.com/pharmaverse/admiral/discussions/2330
